### PR TITLE
vagrantfile for windows

### PR DESCRIPTION
**Pull request by Anakonda**
-Fetch all that is needed for building borg.
-Build windows standalone exe.

-~~Todo: try to use inno setup automatically~~
-~~Make base box~~ Used one from atlas

It expires in 90 days. You can allways load a snapshot or download new.

### CODE REVIEW

**Comment by ThomasWaldmann** [comment]
See the tickets/bounties on the issue tracker about this.

**Comment by Anakonda** [comment]
I'll take a look.
I thought someone checked "working on it" in bountysource.

**Comment by ThomasWaldmann** [comment]
Right, someone began working on that 24d ago.
Also, I somehow confused it by assuming appveyor uses vagrant.

File: Vagrantfile [hunk 1]
```diff
@@ -272,22 +272,50 @@ def fix_perms
...
   config.vm.provider :virtualbox do |v|
     #v.gui = true
     v.cpus = 1
   end
 
   # Linux
   config.vm.define "centos7_64" do |b|
+    config.vm.synced_folder ".", "/vagrant/borg/borg", :type => "rsync", :rsync__args => ["--verbose", "--archive", "--delete", "-z"]
+    config.vm.synced_folder ".", "/vagrant", disabled: true
+    config.vm.provision "fix perms", :type => :shell, :inline => fix_perms
```
Comment (ThomasWaldmann) [inline Vagrantfile]: why is it needed to modify the linux VM definitions to add windows support?

Comment (RonnyPfannschmidt) [inline Vagrantfile]: at first glance this shouldnt use `config.vm`

Comment (Anakonda) [inline Vagrantfile]: I didn't find proper way to do it, and yes it should be `b.vm`

File: Vagrantfile [hunk 2]
```diff
@@ -313,9 +313,9 @@ Vagrant.configure(2) do |config|
```
Comment (enkore) [inline Vagrantfile]: This commit looks like it fixes a (cosmetic or more serious?) issue primarily for the master branch. Can you cherry-pick it towards master?

Comment (ThomasWaldmann) [inline Vagrantfile]: I only see a slight fix for Vagrantfile changes introduced in windows branch - why should they be in master now?

**Comment by Anakonda** [comment]
I'm giving up trying to make installer automatically. Installing the inno setup itself isn't straightforward enough and our installer isn't that useful.

**Comment by enkore** [comment]
Does this work for you? Is any special setup required, since this doesn't work for me, I get this after ten minutes or so

```
   windows: WinRM address: 127.0.0.1:55985
    windows: WinRM username: vagrant
    windows: WinRM execution_time_limit: PT2H
    windows: WinRM transport: plaintext
    windows: Warning: Authentication failure. Retrying...
    windows: Warning: Authentication failure. Retrying...
    windows: Warning: Authentication failure. Retrying...
    windows: Warning: Authentication failure. Retrying...
    windows: Warning: Authentication failure. Retrying...
    windows: Warning: Authentication failure. Retrying...
    windows: Warning: Authentication failure. Retrying...
Timed out while waiting for the machine to boot. This means that
Vagrant was unable to communicate with the guest machine within
the configured ("config.vm.boot_timeout" value) time period.

If you look above, you should be able to see the error(s) that
Vagrant had when attempting to connect to the machine. These errors
are usually good hints as to what may be wrong.

If you're using a custom box, make sure that networking is properly
working and you're able to connect to the machine. It is a common
problem that networking isn't setup properly in these boxes.
Verify that authentication configurations are also setup properly,
as well.

If the box appears to be booting properly, you may want to increase
the timeout ("config.vm.boot_timeout") value.
```

**Comment by Anakonda** [comment]
I'm spending summer somewhere with slow internet so I'm not checking that anytime soon :(
Loging in to WinRM needs UAC to be turned of from registry that *should* be done on these boxes.

**Pull request merged**